sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/environment.py

```diff
@@ -317,13 +321,14 @@ def __eq__(self, other) -> bool:
     def __ne__(self, other) -> bool:
         return not self.__eq__(other)
 
-    def _generate_anonymous_name_version(self, source: str, conda_file: str = None):
+    def _generate_anonymous_name_version(self, source: str, conda_file: str = None, inference_config: Dict = None):
         hash_str = ""
         if source == "image":
-            if not conda_file:
-                hash_str = hash_str.join(get_md5_string(self.image))
-            else:
-                hash_str = hash_str.join(get_md5_string(self.image)).join(get_md5_string(conda_file))
+            hash_str = hash_str.join(get_md5_string(self.image))
+            if inference_config:
+                hash_str = hash_str.join(get_md5_string(yaml.dump(inference_config, sort_keys=True)))
+            if conda_file:
```

Comment:
Let me confirm this - the order of considerations is hash of conda > inference > image? (not using `elif` is intentional)? If so, can we codify this order of precedence in a test, so that it cannot be easily broken?

Reply:
Please see the test case added at the line number 93, it has an environment which has all three and I am checking the  the hash says the same. Does this answer the test question?

Reply:
I think that works - thanks!